model/task_history.go

```diff
@@ -389,35 +391,20 @@ func (self *taskHistoryIterator) GetDistinctTestNames(ctx context.Context, env e
 			{"$unwind": fmt.Sprintf("$%v", testResultsKey)},
 			{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", testResultsKey, task.TestResultTestFileKey)}},
 		},
-		opts,
 	)
-	if err != nil {
-		return nil, errors.Wrapf(err, "error getting cursor")
-	}
-	if cursor == nil {
-		return nil, errors.New("nil cursor returned")
-	}
-	defer cursor.Close(ctx)
-
-	names := []string{}
-	for cursor.Next(ctx) {
-		select {
-		case <-ctx.Done():
-			return nil, errors.New("context cancelled")
-		default:
-			res := bson.M{}
-			if err := cursor.Decode(&res); err != nil {
-				return nil, errors.Wrapf(err, "error decoding result")
-			}
-			// remove quotes from value
-			val := strings.Replace(res["_id"].(string), "\"", "", -1)
-			names = append(names, val)
+	pipeline.MaxTime(time.Minute)
+
+	var names []string
+	res := bson.M{}
+	iter := pipeline.Iter()
+	for iter.Next(&res) {
+		names = append(names, res["_id"].(string))
+		if ctx.Err() != nil {
+			_ = iter.Close()
+			return nil, errors.Wrap(ctx.Err(), "iterating results")
 		}
 	}
-	if err := cursor.Err(); err != nil {
-		return nil, errors.Wrap(err, "error reading cursor")
-	}
-	sort.Strings(names)
```

Comment:
Same question about this sort.

Reply:
This was another instance of the same thing: the changes from #4500 made them always come back in a different order, [for some reason](https://github.com/evergreen-ci/evergreen/pull/4500#issuecomment-802322659). When I reverted it the sort was removed too.

When I get a hold of staging I'll see if this is still an issue.

Reply:
ah, I'm wrong: Annie said it happened before too.
I'll add it back

Reply:
done